Adds stub project detail component

naomiyaki opened the pull request.

Commit:
[F] Add hero component in project detail view

[FEATURE #108401192] The project detail has an initial hero area that
renders maker names, title, description, and product image. There are
also stub links to access the project in reader and buy the print
version.
TODO: Move the hero are into its own component, as well as the makers
figure.

Commit:
[F] Refactor project detail into smaller files

[FEATURE DELIVERS #108401192] The project detail hero is moved into
a simpler component that retrieves a project and its makers from the
project detail container. In addition, many svgs have been added to
support other elements that are forthcoming in the project detail view
functionality.

Commit:
[F] Add stub activity to project detail view

[FEATURE DELIVERS #108401236] Two components have been added to list
and render activity updates shown in the project detail view in the
comps. The list component accepts what is currently mock data from the
project detail container, and contains logic to match the heights in a
list of activity tiles.
The tile container itself contains several layout variants that will
be loaded depending on the "type" value of the update. Note that the
logic for determining and passing update types is entirely for markup
construction purposes at this time, and the data structures for these
activity updates are likely to change.
Currently this activity can be shown in its entirety or hidden based
on a minimum count declared in the project detail container.
The activity updates contain logic to center their content vertically
using padding calculated based on their parent list item height. Note
that this logic runs only once when the components load/are updated,
and it's not yet a final, responsive solution.

Commit:
[F] Add stub component for project texts

[FEATURE DELIVERS #108401254] The static markup for the texts
component tries to approximate every permutation of a text thumbnail
output, but it is 10% static and not make any guesses about the data
structure from the API. When it is, this component will likely need
refactoring into smaller blocks of layout logic.
Like the hero, this is a initial responsive, but not mobile ready
layout.

Commit:
[F] Add stub component for project meta

[FEATURE DELIVERS #108401580] The project detail container passes a
very basic set of sample key-value data to a project meta listing
component. This can be further abstracted if the key/value markup needs
to be used elsewhere, but for simplicity it stubbed as one component.

zdavis commented on client/src/components/frontend/ProjectActivityUpdate.js:
```diff
@@ -0,0 +1,122 @@
+  };
+
+  renderUpdateContent = (update)=> {
+    switch (update.type) {
```
Instead of a big switch statement, I'd like to see a switch that calls out to individual functions. Eg, this switch could call renderUpdateComment, renderUpdateFile, etc.

I'm also not crazy about thinking of each boxes as "updates." That word is too fuzzy. Each individual box will represent a project event—a thing that happened in relation to the project. "Updates" is too specific, as this activity can be something external to the project, eg, a tweet with a specific hash tag, rather than an update to the project itself.

I'd call the component EventGrid or EventList. It can be passed any collection of event and render them. If we later decide that texts can have events too, then we can easily repurpose the component for that purpose. Cool?

zdavis commented on client/src/components/frontend/ProjectMeta.js:
```diff
@@ -0,0 +1,27 @@
+import React, { Component, PropTypes } from 'react';
+
+export default class ProjectMeta extends Component {
```
When we choose names, we can inadvertently create a coupling between things that shouldn't be coupled. While this component is currently used on a project page, there's no reason why it couldn't be used on another model that also has meta data. By including project in the name, we imply a coupling that doesn't need to exist. Along these lines, I'd rename it to MetaAttributes, I think.

zdavis commented on client/src/components/frontend/ProjectTexts.js:
```diff
@@ -0,0 +1,191 @@
+import React, { Component, PropTypes } from 'react';
+import { Link } from 'react-router';
+
+export default class ProjectTexts extends Component {
```
As above, this should be called GroupedTexts or something to that effect. The list is a list of texts grouped by tag. While that list is appearing in the context of the project, there's no need to codify that into the component name.

zdavis commented on client/src/components/frontend/ProjectTexts.js:
```diff
@@ -0,0 +1,191 @@
+
+  static propTypes = {};
+
+  render = () => {
```
Doesn't necessarily need to happen in this pull request, but even without real data, this component could be broken up into smaller pieces. The GroupedTexts components should take a collection of texts and chunk them by tag. It should loop through each tag chunk, render the tag, then loop through each text and render the text (thumbnail, title, stats, etc). The bit that renders the text should be its own component.

Commit:
[F] Refactors event list and meta attributes

Events list and meta attributes have been given more general, flexible
names. The events list component has also been refactored into a switch
statement that picks discrete functions by event type instead of
containing all logic within the switch.

Commit:
[F] Project texts are abstracted into components
Previously, the project texts component contained entirely static html.
After some internal discussion at CIC, the data structure for texts has
been loosely modeled in the project detail container and used to
simulate more structured components, where the rendering of each text
thumbnail is handled by single-use component with separate components
for project list outputs.
